web/packages/teleport/src/DesktopSession/KeyboardHandler.tsx

```diff
@@ -0,0 +1,271 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Platform, getPlatform } from 'design/platform';
+
+import { TdpClient, ButtonState } from 'teleport/lib/tdp';
+import { SyncKeys } from 'teleport/lib/tdp/codec';
+
+/**
+ * Handles keyboard events.
+ */
+export class KeyboardHandler {
+  /**
+   * Cache for keys whose down event has been withheld until the next keydown or keyup,
```

Comment:
nit: I wouldn't call these caches. A cache is generally a performance optimization that replaces slow lookups with fast one.

This is really just a lookup table, not a cache for some slower downstream store.